core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -65,14 +65,16 @@ public void available() {
                 }
 
                 if (defaultDb.equals(db.databaseName())) {
-                    final List<String> versions = db.executeTransactionally("CALL dbms.components", Collections.emptyMap(),
-                            r -> (List<String>) r.next().get("versions"));
-                    final String apocFullVersion = Version.class.getPackage().getImplementationVersion();
-                    if (isVersionDifferent(versions, apocFullVersion)) {
-                        userLog.warn("The apoc version (%s) and the Neo4j DBMS versions %s are incompatible. \n" +
-                                        "See the compatibility matrix in https://neo4j.com/labs/apoc/4.4/installation/ to see the correct version",
-                                apocFullVersion, versions.toString());
-                    }
+                    try {
+                        final List<String> versions = db.executeTransactionally("CALL dbms.components", Collections.emptyMap(),
+                                r -> (List<String>) r.next().get("versions"));
+                        final String apocFullVersion = Version.class.getPackage().getImplementationVersion();
+                        if (isVersionDifferent(versions, apocFullVersion)) {
+                            userLog.warn("The apoc version (%s) and the Neo4j DBMS versions %s are incompatible. \n" +
+                                            "See the compatibility matrix in https://neo4j.com/labs/apoc/4.4/installation/ to see the correct version",
+                                    apocFullVersion, versions.toString());
+                        }
+                    } catch (Exception ignored) {}
```

Comment:
this is not related to this PR but I found some exceptions during the shutdown of the test database in the `MetaTest` class.
I can revert it if you want

Reply:
ignored exceptions are not a great practice.  Recommend changing to (minimum) logging that exception.  This way you can address the test database issue without swallowing an error someone might need to know about.

Reply:
I agree, that we shouldn't do this

Reply:
@moxious @Lojjs 
I ignored it because the stack trace is this one:
```bash
org.neo4j.graphdb.TransactionTerminatedException: The transaction has been terminated. Retry your operation in a new transaction, and you should see a successful result. The database is not currently available to serve your request, refer to the database logs for more details. Retrying your request at a later time may succeed. 
    	at org.neo4j.kernel.impl.api.KernelTransactionImplementation.assertOpen(KernelTransactionImplementation.java:688)
    	at org.neo4j.kernel.impl.newapi.AllStoreHolder.callProcedure(AllStoreHolder.java:1074)
    	at org.neo4j.kernel.impl.newapi.AllStoreHolder.procedureCallDbms(AllStoreHolder.java:1025)
    	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.$anonfun$callDbmsProcedure$1(CallSupport.scala:59)
    	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callProcedure(CallSupport.scala:70)
    	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callDbmsProcedure(CallSupport.scala:59)
    	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundReadQueryContext.callDbmsProcedure(TransactionBoundQueryContext.scala:1147)
    	at org.neo4j.cypher.internal.planning.ExceptionTranslatingReadQueryContext.callDbmsProcedure(ExceptionTranslatingQueryContext.scala:235)
    	at org.neo4j.cypher.internal.runtime.DbmsCallMode$.callProcedure(ProcedureCallMode.scala:87)
    	at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe.call(ProcedureCallPipe.scala:88)
    	at org.neo4j.cypher.internal.runtime.interpreted.pipes.ProcedureCallPipe.$anonfun$internalCreateResultsByAppending$1(ProcedureCallPipe.scala:74)
    	at org.neo4j.cypher.internal.runtime.ClosingIterator$$anon$1.nextCur(ClosingIterator.scala:107)
    	at org.neo4j.cypher.internal.runtime.ClosingIterator$$anon$1.innerHasNext(ClosingIterator.scala:113)
    	at org.neo4j.cypher.internal.runtime.ClosingIterator.hasNext(ClosingIterator.scala:93)
    	at org.neo4j.cypher.internal.runtime.ClosingIterator$$anon$3.innerHasNext(ClosingIterator.scala:152)
    	at org.neo4j.cypher.internal.runtime.ClosingIterator.hasNext(ClosingIterator.scala:93)
    	at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.serveResults(PipeExecutionResult.scala:85)
    	at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.request(PipeExecutionResult.scala:73)
    	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:90)
    	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.fetchResults(ResultSubscriber.java:362)
    	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.materialize(ResultSubscriber.java:92)
    	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.initiate(StandardInternalExecutionResult.scala:65)
    	at org.neo4j.cypher.internal.result.ClosingExecutionResult.$anonfun$initiate$1(ClosingExecutionResult.scala:58)
    	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
    	at org.neo4j.cypher.internal.result.ClosingExecutionResult.safely(ClosingExecutionResult.scala:99)
```

If we log it we have to be very careful about the message that we return because otherwise it could confuse the users.
What about `Cannot check APOC version compatibility because of a transient error. Retrying your request at a later time may succeed`?